std/digest/hmac.d

```diff
@@ -258,15 +261,15 @@ if (isDigest!H)
     DigestType!H hmac(T...)(scope T data, scope const(ubyte)[] secret)
     if (allSatisfy!(isDigestibleRange, typeof(data)))
     {
-        import std.algorithm.mutation : copy;
+        import std.range : put;
         auto hash = HMAC!(H, blockSize)(secret);
         foreach (datum; data)
-            copy(datum, &hash);
+            put(hash, datum);
```

Comment:
None, we even save a function call on bad optimizers as [copy](https://github.com/dlang/phobos/blob/681c74357af21acd6cbfecbf26915619b8a60e65/std/algorithm/mutation.d#L399) does _exactly_ the same:

```d
TargetRange copy(SourceRange, TargetRange)(SourceRange source, TargetRange target)
if (!areCopyCompatibleArrays!(SourceRange, TargetRange) &&
    isInputRange!SourceRange &&
    isOutputRange!(TargetRange, ElementType!SourceRange))
{
    // Specialize for 2 random access ranges.
    // Typically 2 random access ranges are faster iterated by common
    // index than by x.popFront(), y.popFront() pair
    static if (isRandomAccessRange!SourceRange &&
               hasLength!SourceRange &&
               hasSlicing!TargetRange &&
               isRandomAccessRange!TargetRange &&
               hasLength!TargetRange)
    {
        auto len = source.length;
        foreach (idx; 0 .. len)
            target[idx] = source[idx];
        return target[len .. target.length];
    }
    else
    {
        put(target, source);
        return target;
    }
}
```

The output range`HMAC` doesn't have a `length`, nor slicing or random access.